File: Program's_Contributed_By_Contributors/Java_Programs/Sorting_Algorithms/Quick_Sort.java
public class Quick_Sort {
    public int partion(int[] nums, int low, int high) {
        int i = low, j = high, pivot = nums[low];
        while (i < j) {
            while (nums[i] <= pivot && i <= high - 1)
                i++;
            while (j >= low + 1 && nums[j] > pivot)
                j--;
            if (i < j) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            }
        }
        int temp = nums[low];
        nums[low] = nums[j];
        nums[j] = temp;
        return j;

    }

    public void quicksort(int[] nums, int low, int high) {
        if (low < high) {
            int p = partion(nums, low, high);
            quicksort(nums, low, p - 1);
            quicksort(nums, p + 1, high);
        }
    }

    public int[] sortArray(int[] nums) {
        quicksort(nums, 0, nums.length - 1);
        return nums;
    }
}

File: C-Sharp/sort_algorithms/quick_sort.cs
using System;

class QuickSort
{
    public static void Main(string[] args)
    {
        int[] arr = { 12, 11, 13, 5, 6, 7 };

        Console.WriteLine("Original Array:");
        PrintArray(arr);

        QuickSortAlgorithm(arr, 0, arr.Length - 1);

        Console.WriteLine("\nSorted Array:");
        PrintArray(arr);
    }

    public static void QuickSortAlgorithm(int[] arr, int low, int high)
    {
        if (low < high)
        {
            int partitionIndex = Partition(arr, low, high);

            QuickSortAlgorithm(arr, low, partitionIndex - 1);
            QuickSortAlgorithm(arr, partitionIndex + 1, high);
        }
    }

    public static int Partition(int[] arr, int low, int high)
    {
        int pivot = arr[high];
        int i = low - 1;

        for (int j = low; j < high; j++)
        {
            if (arr[j] < pivot)
            {
                i++;
                Swap(arr, i, j);
            }
        }

        Swap(arr, i + 1, high);
        return i + 1;
    }

    public static void Swap(int[] arr, int i, int j)
    {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void PrintArray(int[] arr)
    {
        foreach (var num in arr)
        {
            Console.Write(num + " ");
        }
        Console.WriteLine();
    }
}
